webapp/advantage/views.py

```diff
@@ -757,6 +757,17 @@ def accept_renewal(renewal_id):
     return g.api.accept_renewal(renewal_id)
 
 
+@advantage_decorator(permission="user", response="json")
+def get_account_offers():
+    try:
+        account = g.api.get_purchase_account("canonical-ua")
+    except UAContractsUserHasNoAccount:
```

Comment:
If the user has `technical` role `get_purchase_account` throws a `AccessForbiddenError`. We have to `except` that too and throw a `403`. 

You can see the error rules of an endpoint at `error_rules=["default", "no-found", "user-role"]` in `api.py`, not saying it's the smartest thing in the world, but it's attempting to gather most of the error handling logic in one place.